### PR TITLE
catch exception due to subgraph removal

### PR DESCRIPTION
**Pull request by madumas**
Potential fix to issue #19

### CODE REVIEW

**Comment by aasseman** [comment]
Thanks! This fixes a pretty big omission indeed.
However, it doesn't look like it's going to help with #19 , as that one is related to https://github.com/semiotic-ai/autoagora/blob/9bb4903db4922a53b013d1cb8e98394d03129d37/autoagora/query_metrics.py#L60 not being caught by the backoff: https://github.com/semiotic-ai/autoagora/blob/9bb4903db4922a53b013d1cb8e98394d03129d37/autoagora/query_metrics.py#L54-L56

**Comment by aasseman** [comment]
I was working on setting up contribution conventions, and finally got them merged in #24 . Sorry if I catch you off guard here :/

Could you amend your commit to have a [conventional message](https://www.conventionalcommits.org/en/v1.0.0/#summary) (it would be a `fix: [...]` in this case), and `--signoff` if you agree with the [DCO](https://developercertificate.org/). 

There's some more details you might be interested in in the new `CONTRIBUTING.md`.

Otherwise LGTM!

**Comment by madumas** [comment]
should be good to go now

File: autoagora/price_multiplier.py [hunk 1]
```diff
@@ -104,6 +104,8 @@ async def price_bandit_loop(subgraph: str):
             loss = bandit.update_policy()
             if loss is not None:
                 logging.debug("Price bandit %s - Training loss: %s", subgraph, loss)
+    except asyncio.CancelledError:
+        logging.debug("Price bandit %s - Removing bandit loop", subgraph)
```
Comment (aasseman) [inline autoagora/price_multiplier.py:108]: I checked [the official docs](https://docs.python.org/3/library/asyncio-task.html#task-cancellation), and it says *not* to suppress `CancelledError`, and to re-raise it if it's been caught. Thanks for making me look more carefully at that piece of code!

```suggestion
    except asyncio.CancelledError as cancelledError:
        logging.debug("Price bandit %s - Removing bandit loop", subgraph)
        raise cancelledError
```

**Review by aasseman — CHANGES_REQUESTED** [review]
There's some bug in GitHub, somehow the tests wouldn't trigger on the PR. However they ran on your fork!
Tests failed because `asyncio` needs to imported in `price_multiplier.py`.

File: autoagora/price_multiplier.py (hunk 1 above)
Comment (madumas) [reply autoagora/price_multiplier.py]: ok, added the raise as instruction. no unwanted effect

**Comment by madumas** [comment]
added the missing raise and import

**Review by aasseman — APPROVED** [review]

**Pull request merged**